firestore/src/main/java/com/firebase/ui/firestore/paging/FirestorePagingOptions.java

```diff
@@ -66,6 +66,19 @@ public LifecycleOwner getOwner() {
         private LifecycleOwner mOwner;
         private DiffUtil.ItemCallback<DocumentSnapshot> mDiffCallback;
 
+        /**
+         * Directly set the {@link DocumentSnapshot}.
+         * <p>
+         * Do not call this method after calling {@code setQuery}.
+         */
+        @NonNull
+        public Builder<T> setDocumentSnapshot(@NonNull LiveData<PagedList<DocumentSnapshot>> data,
```

Comment:
A few things here:

1) I think this method should be called `setData()`
2) I think we need to offer two overloads, one that takes `Class<T> modelClass` (like you have) and another that takes `SnapshotParser<T> parser` so that it's as expressive as `setQuery()`